src/config.rs

```diff
@@ -61,8 +61,68 @@ struct Root {
     clouds: Clouds,
 }
 
-fn find_config() -> Option<PathBuf> {
-    let current = Path::new("./clouds.yaml");
+/// Merge two nested serde_yaml::Mapping structs
+/// The values from src are merged into dest. Values in src override values in dest.
+fn merge_mappings(src: &serde_yaml::Mapping, dest: &mut serde_yaml::Mapping) {
```

Comment:
It feels like we could take `src` by value and avoid `to_owned` later on (which likely results in cloning).

Reply:
I'm having some trouble with this one. I obtain the Mappings in the calling code by calling the as_mapping() (or as_mapping_mut()) method on the serde_yaml::Value enum which returns a Option<&Mapping> which I then have to somehow convert to an owned Mapping. I always end up calling some method that does a clone behind the scenes. It just feels like I'm moving the cloning from merge_mappings to the calling code (maybe that's better?).

I'm still a bit new to rust (as you might have guessed) and maybe I missed something in the docs.

Reply:
`Value` is an enum, so you should be able to do something like
```rust
if let serde_yaml::Value::Mapping(mapping) = value {

}
```

Reply:
Note that I don't insist on you making this change, I can follow up. My only concern with this patch now is the failing unit test.

Reply:
Hi!

I got all the way here:|

```
fn merge_mappings(src: serde_yaml::Mapping, dest: &mut serde_yaml::Mapping) {
    for (src_key, src_value) in src.into_iter() {
        if let (serde_yaml::Value::Mapping(src_mapping), Some(dest_value)) =
            (src_value, dest.get_mut(&src_key))
        {
(...)

        let _ = dest.insert(src_key, src_value);
}
```

which fails because `src_value` is moved in the `if let` and then inserted in `dest`. I'm kind of stuck here. My next step was to look at re-implementing `merge_mappings()` so it does the merge non-recursively but I kind of liked the recursion in this function. What do you think?

Reply:
I guess you could do it with `match` instead of `if let`. I don't want to hold this work more than necessary, so if this patch is good to go, I can follow-up (I'll ping you if I do).